autopilot/autopilot.go

```diff
@@ -134,10 +134,10 @@ func (ap *Autopilot) Run() error {
 		// update contractor's internal state of consensus
 		//
 		// TODO: this should be done through a pub/sub mechanism on the bus
-		ap.c.applyConsensusState(cfg, cs)
+		ap.c.ApplyConsensusState(cfg, cs)
```

Comment:
`nit` would've preferred a rename of `currentPeriod` to `currPeriod` as opposed to exposing what would be exposed if the contractor were in its own package. It's consistent now but it's hard to keep consistent and it's just weird to expose methods that don't need to be exposed

Reply:
I renamed it but tbh I feel like that only leaves us with nasty options.
A: field is currPeriod and method is currentPeriod. It's not obvious from the name when you want to choose one or the other.
B: use c.currentPeriod and c.mu directly in the autopilot loop. Big no no in my opinion to access a type's mutex from outside the type's own method.

I know that we don't like deep folder hierarchies but it would be quite clean to simply add a `contractor` package. Then we are required to expose the methods called by the parent (the autopilot). It's usually also implied that exposed methods are thread-safe so it would all add up.

Reply:
Definitely see your point, I just think the exposed methods weren't the right approach. Maybe renaming the field to `periodHeight` makes it less nasty. That way the `contractor` has a `blockHeight` and `periodHeight`, the getter can be called `currentPeriod` and we don't needless add packages. That way we can avoid adding extra packages as long as we can 🤷🏻‍♂️ if the `contractor` is a package then I think the `migrator` needs to be one also and so on..